vllm/distributed/device_communicators/tpu_communicator.py

```diff
@@ -22,11 +22,24 @@
     import torch_xla.core.xla_model as xm
     import torch_xla.runtime as xr
     from torch_xla._internal import pjrt
+    from torch_xla.distributed.xla_multiprocessing import (
+        create_optimized_replica_groups)
 
     if USE_RAY:
         from vllm.executor import ray_utils
 
 
+@torch.library.custom_op("tpu::all_reduce", mutates_args=())
```

Comment:
is it possible to use custom op call in vllm directly? e.g. extending this line https://github.com/vllm-project/vllm/blob/44f990515b124272f87954fc763d90697d8aa1db/vllm/distributed/parallel_state.py#L222 to include tpu.

i didn't do it initially, because i remember tpu has some custom dynamo-related logic.

Reply:
Thanks for you suggestion!

I removed the custom op in `tpu_communicator.py` and made use of the custom op for TPU in `parallel_state.py`